index.html

```diff
@@ -0,0 +1,201 @@
+  classDef file fill:none,color:#eee;
+  style proc fill:none,stroke:fuchsia,color:fuchsia;
+```
+
+A common case is that of a processor which consumes an input schema and generates an output schema.
+
+The general guidance for processor behavior is: don't react to what you don't understand.
+
+Specifically, processors:
+  - **should** pass through `@using` directives which reference unknown spec URLs
+  - **should** pass through unknown directives, types, and other schema elements
+  - **should** ignore meaningless URL components (such as unknown query parameters or fragments) within `@using` URLs
+
+An exception to this is processors which prepare the schema for final public consumption. Such processors **may** choose to eliminate all unknown directives and prefixed types in order to hide schema implementation details within the published schema. This will impair the operation of tooling which relies on these directives—such tools will not be able to run on the output schema, so the benefits and costs of this kind of information hiding should be weighed carefully on a case-by-case basis.
+
+## Spec URLs
+
+```graphql
+#::[Declaration of using_URL scalar]
+scalar using_URL
+  @specifiedBy(url: "https://apollographql.com/docs/specs/using#spec-urls")
+```
+
+The URL of a specification.
+
+Specification URLs **must** be valid [RFC 3986 URLs](https://tools.ietf.org/html/rfc3986). They **should** provide the content of the appropriate version of the specification in some human-readable form. It is not required that the content be machine-readable in any particular way.
+
+The final component of the URL's `"/"`-separated `path` **must** be a valid [SemVer specifier](#semver-specifiers) identifying the version of the spec in use. Everything prior to the version specifier is the spec's *base URL*. Processors **should** treat the spec's base URL as an opaque string identifying the specification.
+
+This spec assigns no meaning to additional URL components, such as the query string or fragment. Processors **should** ignore any meaningless URL components when parsing the base URL and semver specifier.
+
+```graphql
+#::[`@using` directive with a query string]
+schema
+  @using(spec: "https://spec.example.com/exampleSpec/1?unrelated=value")
+  { ... }
+```
+
+This directive:
+- Declares that the document conforms to a spec which can be found at `https://spec.example.com/exampleSpec/1`
+- Has a base URL of `https://spec.example.com/exampleSpec`
+- Has a [semver specifier](#semver-specifier) of `1`
+- Has a query string of `"unrelated=value":
+  - This has no effect unless `exampleSpec` assigns it specific meaning.
+
+## Semantic Versioning
+
+Specifications **should** provide a semantic version number.
+
+### SemVer Specifiers
+
+The final `/`-separated path component **must** be a semver specifier. Semver specifiers take the form `v*major*.*minor*.*patch*-*prerelease*`. The minor, patch, and pre-release specifiers are all optional. Examples:
```

Comment:
- This renders poorly (the asterisks are rendered)
- Should specify legal characters for each portion, perhaps by reference to SemVer 2
- Probably should be explicit (though the example is clear) that leaving out the specifier also means leaving out the character before it
- Probably should be explicit that eg `v1-foo` is not legal
- Probably should be even more explicit that the `v` is required

(Ah, OK, several of these concerns are resolved by section 6.4. Perhaps that section could just be rolled into 6.1?)

Reply:
Rendering is still bad, and I still think it would be helpful to at least have a forward reference to the BNF section (or just roll it into the definition).

Reply:
yes, i have yet to address this comment, that's why the github convo is still open ;)

Reply:
ah, gotcha! Didn't realize that's what you were using it for. GitHub conversations can be a challenge to follow.